Fin 48 add account creation

ArchieSW opened the pull request.

ArchieSW committed:
Add account model

Add account physical and logical model

Jira ID: FIN-48
Testing done: ran existing and new tests

ArchieSW committed:
Add account creation endpoint

Add account creation endpoint with creation usecase

Jira ID: FIN-48
Testing done: ran existing and new tests

vat78 commented on src/main/java/ru/finess/finess/identity/infrastructure/SecurityConfig.java:
```diff
@@ -117,6 +118,23 @@ public Supplier<Optional<User>> currentUser() {
   }
 
+  @Bean
+  public Supplier<User> currentUserSupplier() {
```
Confusing naming because of existing method Supplier<Optional<User>> currentUser().
I would prefer to use only one method, with Optional, and throw an exception on the client side.

ArchieSW replied:
Removed this method, left only with Optional. Changed the return type from `User` to `UserId` so that the database call would be called on the client side and there would be fewer "surprises", thx

vat78 commented on src/main/java/ru/finess/finess/payment/application/AccountCreationUseCase.java:
```diff
@@ -0,0 +1,45 @@
+  @Override
+  public Result<Account, Void> execute(@NonNull Parameters parameters) {
+    Account result =
+        transactionWrapper.execute(
```
What's the point of adding wrapping to a transaction with only one operation?

ArchieSW replied:
Initially, there was a need for an easy change of framework (infrastructure) in the future, but now the need has disappeared, so I changed to regular Spring annotations, thx

ArchieSW committed:
Code refactoring

Rename classes and variables to align code style, moved session
to domain, added lombok where it looks fine and removed
TransactionWrapper since since it is unreadable and does not make
the application layer cleaner from an architectural point of view

Jira ID: FIN-48
Testing done: ran existing tests
